### PR TITLE
Added support for specifying a separate project to run bq jobs

### PR DESCRIPTION
**Pull request by monicavalluri**
Add support for specifying a different project for running BQ jobs. This prevents granting Security Admin in the project that the data resides in. 
This PR has changes from https://github.com/GoogleCloudPlatform/healthcare-federated-access-services/pull/9

### CODE REVIEW

**Comment by monicavalluri** [comment]
@chaopeng could you please help me with the build logs, I do not have permissions to view the logs.

**Comment by chaopeng** [comment]
I also don't have permission. I think this is not build logs.

File: lib/saw/warehouse_test.go [hunk 1]
```diff
@@ -415,26 +468,61 @@ type fakeBQ struct {
...
+		Access: []*bigquery.DatasetAccess{},
+	}
+	return bq, f.getResponseErr
 }
 
 func (f *fakeBQ) Set(ctx context.Context, project string, dataset string, ds *bigquery.Dataset) error {
+	for _, bqd := range ds.Access {
+		a := &bigquery.DatasetAccess{}
+		a.Role = bqd.Role
+		a.UserByEmail = bqd.UserByEmail
```
Comment (cdvoisin) [inline lib/saw/warehouse_test.go:489]: Can rewrite the creation of this object (above lines) as:

a := &bigquery.DatasetAccess{
  Role: bqd.Role,
  UserByEmail: bqd.UserByEmail,
}

That is, what is here is valid and sometimes useful if there are conditional statements to doing this kind of assignment, but normally items are just added to the object allocator when it is a straight-up assignment to improve readability.

File: lib/saw/warehouse_test.go [hunk 2]
```diff
@@ -415,26 +468,61 @@ type fakeBQ struct {
...
+	}
+	return policy, f.getResponseErr
 }
 
 func (f *fakeCRM) Set(ctx context.Context, project string, policy *cloudresourcemanager.Policy) error {
+	f.crmState.Project = project
+	for _, binding := range policy.Bindings {
+		b := &cloudresourcemanager.Binding{}
+		b.Role = binding.Role
+		b.Members = binding.Members
```
Comment (cdvoisin) [inline lib/saw/warehouse_test.go:524]: ditto

File: lib/saw/warehouse_test.go [hunk 3]
```diff
@@ -362,14 +413,16 @@ func newFix(t *testing.T) (*Fix, func() error) {
...
-		bqds: &fakeBQ{},
-		crm:  &fakeCRM{},
+		bqds: &fakeBQ{bqState: bqState},
+		crm:  &fakeCRM{crmState: crmState},
 		gcs:  &fakeGCS{},
 	}
 	f.rpc, cleanup = fakegrpc.New()
-
+	f.crmState = crmState
+	f.bqState = bqState
```
Comment (cdvoisin) [inline lib/saw/warehouse_test.go:425]: I would probably add these to the allocator above as well:

f := &Fix{
  ...
  crmState: crmState,
  bqState: bqState,
}

That is, the line below (f.iamSrv) is also not ideal as there are no conditional statements or other logic that would need it to be separated from the initializer above.

**Comment by cdvoisin** [comment]
The build logs are cryptic here, with a number of errors of the form:
   failed: invalid character 'p' after top-level value

For example:

Step #3 - "go_test": === RUN   TestConfigHandlers
Step #3 - "go_test": E0514 15:18:24.867239    2046 dam.go:467] failed to sync hydra clients: rpc error: code = Internal desc = DecodeJSON() failed: json.Unmarshal(404 page not found
Step #3 - "go_test": ) failed: invalid character 'p' after top-level value
Step #3 - "go_test": === RUN   TestConfigHandlers/#00
...

Could it be that ") failed" is trying to tell us what failed, but it is getting ")" as the name of the item?

File: lib/saw/warehouse_test.go [hunk 4]
```diff
@@ -415,26 +468,61 @@ type fakeBQ struct {
...
+
+type bqState struct {
+	Access []*bigquery.DatasetAccess
 }
 
 func (f *fakeBQ) Get(ctx context.Context, project string, dataset string) (*bigquery.Dataset, error) {
-	return f.getResponse, f.getResponseErr
+	bq := &bigquery.Dataset{
+		Access: []*bigquery.DatasetAccess{},
+	}
```
[the remark this answers is not in the document]
Comment (mbarkley) [reply lib/saw/warehouse_test.go:481]: If I understand correctly, you think the return statement should be:
```
return &bigquery.Dataset{ Access: f.bqState }, f.getResponseErr
```

[the remark this answers is not in the document]
Comment (mbarkley) [reply lib/saw/warehouse_test.go:481]: And that also we should move assign the `Access` as part of setting up the test fixture?

[the remark this answers is not in the document]
Comment (cdvoisin) [reply lib/saw/warehouse_test.go:480]: Let me rephrase since my last comment was hard to parse. Sorry about that!

You already have an "Access" array defined in "bqState". Is it possible that you should be setting and returning that value, or do you really intend to not use that one and always create a new empty one here on each get call?

**Comment by chaopeng** [comment]
@mbarkley sorry, this PR has been merged. Please start another PR for https://github.com/DNAstack/healthcare-federated-access-services/commit/0d18bf6787832d5fb74f0fe50e94d31c9edf2e55 commit
